OutOfSchool/OutOfSchool.WebApi/Controllers/V1/ApplicationController.cs

```diff
@@ -167,12 +168,64 @@ public async Task<IActionResult> GetByProviderId(Guid providerId, [FromQuery] Ap
 
         var applications = await applicationService.GetAllByProvider(providerId, filter).ConfigureAwait(false);
 
-        if (applications == null || !applications.Entities.Any())
+        return this.GetResultOfGetOperation(applications);
+    }
+
+    /// <summary>
+    /// Get collection of applications, that have a pending status.
+    /// </summary>
+    /// <param name="providerId">Provider id.</param>
+    /// <returns>List of applications.</returns>
+    /// <response code="200">Entities were found by given Id.</response>
+    /// <response code="204">No entity with given Id was found.</response>
+    /// <response code="400">Provider with given id was not found.</response>
+    /// <response code="401">User is unauthorized.</response>
+    /// <response code="500">If any server error occurs.</response>
+    [HasPermission(Permissions.ApplicationRead)]
+    [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(SearchResult<ApplicationDto>))]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    [HttpGet("/api/v{version:apiVersion}/provider/{providerId}/applications/pending")]
+    public async Task<IActionResult> GetPendingApplicationsByProviderId(Guid providerId)
+    {
+        // Find a standard provider by given id
+        var providerStandard = await providerService.GetById(providerId).ConfigureAwait(false);
+
+        // Filter for applications, that is set to "Pending"
+        var filter = new ApplicationFilter()
         {
-            return NoContent();
+            Statuses = new List<ApplicationStatus>()
+            {
+                ApplicationStatus.Pending,
+            },
+        };
+
+        SearchResult<ApplicationDto> applications;
+
+        // if: providerStandard is not null - get an applications and return it
+        // else: find a providerAdmin, get an applications and return it
+        if (providerStandard is not null)
+        {
+            applications = await applicationService.GetAllByProvider(providerId, filter).ConfigureAwait(false);
+        }
+        else
+        {
+            var providerAdminIdStringVersion = providerId.ToString();
+            var providerAdmin = await providerAdminService.GetById(providerAdminIdStringVersion).ConfigureAwait(false);
+
+            // Standard and admin providers were not found by given id
+            if (providerAdmin is null)
+            {
+                return BadRequest($"There is no any provider with given id - {providerId}.");
+            }
+
+            applications = await applicationService
+                                     .GetAllByProviderAdmin(providerAdminIdStringVersion, filter, providerAdmin.ProviderId, providerAdmin.IsDeputy)
+                                     .ConfigureAwait(false);
```

Comment:
(Code style, code outlining) Please use standard outlining - 1 tab for wrapped line of code, not n+1 tab.
I.e. for the current code it should be
```
            applications = await applicationService
                .GetAllByProviderAdmin(....)
                .ConfigureAwait(false);
```
instead of current 
```
            applications = await applicationService
                                     .GetAllByProviderAdmin(....)
                                     .ConfigureAwait(false);
```
Idk, maybe it's your IDE configuration - please check it.

Reply:
It is my bad, i was thinking, that IDE outlining is not correctly, so i added one more tab.

Now it is fixed.